### ISSUE TITLE
"password" and "application" flows are broken

### ISSUE THREAD

**Issue by Naid405** [swagger-api/swagger-ui#3128]
When reporting an issue, please provide the following details:
- swagger-ui version - 3.0.11
- a swagger file reproducing the issue:
```
securityDefinitions: {
        OAuth2: {
            type: "oauth2",
            scopes: {
                "api": "API access"
            },
            flow: "application",
            tokenUrl: "/api/v3/oauth/token"
        }
    },
security: [{"OAuth2": []}]
```

According to spec at https://tools.ietf.org/html/rfc6749:
For "password" flow "username" and "password" should be passed in the body using the "application/x-www-form-urlencoded" and "client_id" and "client_password" should be passed via basic auth.
For "application" flow "client_id" and "client_password" should just be passed via basic auth.

Currently none of the options available for "password" flow provide the ability to authorize the way specification suggests since they don't actually send client credentials the way specification suggests.

Also maybe specification way should be the default one.

**Comment by Naid405** [comment]
Any progress on this?

**Comment by bodnia** [comment]
@Naid405 I will provide you with more info asap

**Comment by bodnia** [comment]
@Naid405 'application' flow is fixed, 'password' flow type 'body-parameter' changed. Fixes are in master

**Comment by Naid405** [comment]
Thank you, @bodnia.
Shouldn't 'body-parameter' be the default auth type? Since this is the 'type' specification suggests.

**Comment by bodnia** [comment]
I think I can make it a default one. I'll notify you here once it's merged

**Comment by bodnia** [comment]
@Naid405 merged to master

**Comment by Naid405** [comment]
Thanks again!

**Comment by Naid405** [comment]
Hi, @bodnia. There's another bug with this now - if you just open the "Authorize" window, type username and password and hit the button it doesn't send the client credentials. Only after I select other type and select "Request Body" back it start sending client credentials.

**Comment by webron** [comment]
@Naid405 can you please file a separate issue for that?

**Comment by Naid405** [comment]
@webron done: #3191

**Comment by frol** [comment]
Here is my thorough summary of the state of OAuth2 Password Flow implementation in Swagger-UI: #3227.
